app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      rescue_from ActiveRecord::RecordNotFound, with: :render_record_not_found
+
+      def index
+        render json: notes, status: :ok, each_serializer: IndexNoteSerializer
+      end
+
+      def show
+        render json: note, status: :ok, serializer: ShowNoteSerializer
+      end
+
+      private
+
+      def all_notes
+        Note.all
+      end
+
+      def filtering_params
+        params[:note_type] = note_type
+        params.permit(%i[note_type])
+      end
+
+      def note_type
+        valid_note_type? ? params[:note_type] : nil
+      end
+
+      def valid_note_type?
+        Note.note_types.keys.include?(params[:note_type])
+      end
+
+      def notes
+        all_notes.where(filtering_params)
```

Comment:
en base a mi comentario anterior, haríamos esto así:
```suggestion
        Note.where(filtering_params)
```